backend/app/user_handler_test.go

```diff
@@ -0,0 +1,1288 @@
+		req.Header.Set("Authorization", "Bearer "+token)
+		req.Header.Set("Content-Type", "application/json")
+		resp := httptest.NewRecorder()
+		router.ServeHTTP(resp, req)
+		assert.Equal(t, http.StatusBadRequest, resp.Code)
+		var result map[string]interface{}
+		err = json.Unmarshal(resp.Body.Bytes(), &result)
+		assert.NoError(t, err)
+		assert.Contains(t, result["error"], "invalid SSH key format")
+	})
+
+	t.Run("Add SSH key with duplicate public key", func(t *testing.T) {
+		app, err := SetUp(t)
+		require.NoError(t, err)
+		router := app.router
+		email := "addsshuser5@example.com"
+		username := "Add SSH User5"
+		user := &models.User{
+			ID:       14,
+			Username: username,
+			Email:    email,
+			Password: []byte("securepassword"),
+			Verified: true,
+		}
+		err = app.handlers.db.RegisterUser(user)
+		assert.NoError(t, err)
+		token := GetAuthToken(t, app, 14, email, username, false)
+		payload1 := map[string]interface{}{
+			"name":       "keyA",
+			"public_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDzy9yGz+CsKhjYB3FLr27SaoPQVi/tOZDZ06LnO7NuVUj0yR3e7IJO26cxs6j7tRAGTrA7choRMlQJdCFQfkDCaAL+31fPSihHhB3kxUTnZymaWgZ6s/JxjI/2/kKcLjxMWpMYTs18ZdRJf1DgoiyTV6yhlxAhWJvMxTtC5++h5+Ir7mHoN5QdrRt5AjKEcTEJjoKC3it4itHz7w45hi4y07kFYIk4HcMGrInh1IC/BriU7xKlwYcP2tp0W4GIraDJoD8OR3cgcYd/AFXSnVDtomCq5MaKBUli6FWLCK7E3+0AtYxxLkQ/zFkPsYSFAGGqVp8uq2hI46d0TxhgcG2EsWiF/2yOjtMdX1ab3Ns23p8Q0l/8JxXn6WT9xhme9eb2v8UjukN0AR8j+hp5xoQuSEgXAxkg4PFEa2seYEcE8xZPOSavuQl4wEAjXH/1BHnRHxrBBWixN2xdclHRAKQRwR+EHg8wDQ0EAAxtoCCAVHOepBrmV0JDxJGHQ8euvbs= test@gmail.com",
+		}
+		payload2 := map[string]interface{}{
+			"name":       "keyA",
+			"public_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDzy9yGz+CsKhjYB3FLr27SaoPQVi/tOZDZ06LnO7NuVUj0yR3e7IJO26cxs6j7tRAGTrA7choRMlQJdCFQfkDCaAL+31fPSihHhB3kxUTnZymaWgZ6s/JxjI/2/kKcLjxMWpMYTs18ZdRJf1DgoiyTV6yhlxAhWJvMxTtC5++h5+Ir7mHoN5QdrRt5AjKEcTEJjoKC3it4itHz7w45hi4y07kFYIk4HcMGrInh1IC/BriU7xKlwYcP2tp0W4GIraDJoD8OR3cgcYd/AFXSnVDtomCq5MaKBUli6FWLCK7E3+0AtYxxLkQ/zFkPsYSFAGGqVp8uq2hI46d0TxhgcG2EsWiF/2yOjtMdX1ab3Ns23p8Q0l/8JxXn6WT9xhme9eb2v8UjukN0AR8j+hp5xoQuSEgXAxkg4PFEa2seYEcE8xZPOSavuQl4wEAjXH/1BHnRHxrBBWixN2xdclHRAKQRwR+EHg8wDQ0EAAxtoCCAVHOepBrmV0JDxJGHQ8euvbs= test@gmail.com",
+		}
+		body1, _ := json.Marshal(payload1)
+		req1, _ := http.NewRequest("POST", "/api/v1/user/ssh-keys", bytes.NewReader(body1))
+		req1.Header.Set("Authorization", "Bearer "+token)
+		req1.Header.Set("Content-Type", "application/json")
+		resp1 := httptest.NewRecorder()
+		router.ServeHTTP(resp1, req1)
+		assert.Equal(t, http.StatusCreated, resp1.Code)
+		body2, _ := json.Marshal(payload2)
+		req2, _ := http.NewRequest("POST", "/api/v1/user/ssh-keys", bytes.NewReader(body2))
+		req2.Header.Set("Authorization", "Bearer "+token)
+		req2.Header.Set("Content-Type", "application/json")
+		resp2 := httptest.NewRecorder()
+		router.ServeHTTP(resp2, req2)
+		assert.Equal(t, http.StatusInternalServerError, resp2.Code)
```

Comment:
shouldn't it be bad request?

Reply:
its returned from this 
```go
	if err := h.db.CreateSSHKey(&sshKey); err != nil {
		log.Error().Err(err).Msg("failed to create SSH key")
		InternalServerError(c)
		return
	}
```

Reply:
I know but duplicate keys is a client problem